pkgs/native_assets_builder/test/data/cyclic_package_1/pubspec_overrides.yaml

```diff
@@ -0,0 +1,3 @@
+dependency_overrides:
+  native_assets_cli:
+    path: ../../../../native_assets_cli/
```

Comment:
How does this work when the package is copied to a temp folder? Ditto all the other test packages that get copied around to temp folders during tests.

Reply:
They now get the same handling as pubspec files.

Reply:
Sweet!

So we just need to make sure that the transitive closure is in the override files right? Which I believe you've done correctly in this PR.